source/adapters/hip/program.cpp

```diff
@@ -28,7 +108,80 @@ ur_result_t ur_program_handle_t_::setBinary(const char *Source, size_t Length) {
   return UR_RESULT_SUCCESS;
 }
 
+ur_result_t ur_program_handle_t_::finalizeRelocatable() {
+#ifndef SYCL_ENABLE_KERNEL_FUSION
+  assert(false && "Relocation only available with fusion");
+  return UR_RESULT_ERROR_UNKNOWN;
```

Comment:
This line will never be reached right? because of the above assert being set to false.

Also it would be better to use a backend specific error that then gets automatically passed to a sycl extension, using `setErrorMessage`, like here: https://github.com/intel/llvm/pull/10953/files

Reply:
> This line will never be reached right? 

In a normal setup, correct. Hence the `false` here.

In order to be reached, you would need to either enable fusion in intel/llvm and actively disable it in the adapter or use the finalize property in some other patch. Either way, you need to actively change something in the build (cmake changes) or add a patch (which if tested should fail CI) to trigger this.

I can add a `setErrorMessage` if you care about a non sycl focus use of the adapter, otherwise I don't think it is that useful.

Reply:
Yeah it's fine to leave it as it is.